dicoogle/src/main/java/pt/ua/dicoogle/server/ControlServices.java

```diff
@@ -50,7 +50,7 @@
     
     private ControlServices()
     {
-        TaskManager taskManager = new TaskManager(4);
+        TaskManager taskManager = new TaskManager(Integer.parseInt(System.getProperty("dicoogle.taskManager.nThreads", "4")));
```

Comment:
Removing this line should be nothing more than an improvement, not a refactor. :)
If it breaks something, for some reason, we can keep it and address the problem separately.